pygsti/algorithms/fiducialselection.py

```diff
@@ -365,7 +552,83 @@ def clean_fid_list(model, circuit_cache, available_fid_list,drop_identities=True
         if fid_string in cleaned_circuit_cache_2:
             cleaned_availableFidList.append(available_fid_list[i])
     
+    return cleaned_availableFidList, cleaned_circuit_cache_2
+    
+#function for cleaning up the available fiducial list to drop identities and circuits with duplicate effects
+#Specialized for the case where our fiducials are promised to correspond to clifford operations
+#in which case we know that the matrices must correspond to permutation matrices which
+#gives us some additional structure to exploit.
+def clean_fid_list_clifford(model, circuit_cache, available_fid_list,drop_identities=True, drop_duplicates=True, eq_thresh= 1e-6):
```

Comment:
This might look cleaner if clean_fid_list just took the assume_clifford flag, and switched on it inside the drop_duplicates block.

Reply:
Sounds good. I refactored the deduping code implementing this change.